subiquity/server/controllers/filesystem.py

```diff
@@ -847,6 +854,40 @@ async def _probe(self, *, context=None):
 
     def run_autoinstall_guided(self, layout):
         name = layout['name']
+
+        if name == 'hybrid':
+            if self._system is None:
+                raise Exception(
+                    "can only use name: hybrid when installing core boot "
+                    "classic")
+            encrypted = layout.get('encrypted', None)
+            safety = self._system.storage_encryption.storage_safety
+            support = self._system.storage_encryption.support
+            if encrypted is None:
+                if safety == StorageSafety.ENCRYPTED:
+                    self.use_tpm = True
```

Comment:
Does anyone do a check for `StorageEncryptionSupport.AVAILABLE` in this case?  If we're demanding it but it's not available then maybe that's a fatal error for autoinstall

Reply:
This is confusing, but if storage safety is encrypted and encryption is not available then the encryption support ends up as DEFECTIVE which means that _core_boot_classic_error gets set which means that we get an exception raised before we get here. I'll add a comment about this.